python/lsst/ts/m2/csc.py

```diff
@@ -1151,6 +1212,41 @@ async def _bump_actuator(
 
         await asyncio.sleep(period)
 
+    async def _publish_status_bump_test(self, actuator: int, status: BumpTest) -> None:
+        """Publish the status of actuator bump test.
+
+        Parameters
+        ----------
+        actuator : `int`
+            0-based actuator ID.
+        status : enum `MTM2.BumpTest`
+            Status of the actuator bump test. The current annotation of
+            enum "BumpTest" is to maintain the backward compatibility.
+        """
+
+        # This is to keep the backward compatibility of ts_xml v20.0.0 that
+        # does not have the 'actuatorBumpTestStatus' event defined in xml.
+        # TODO: Remove this after ts_xml v20.1.0.
+        if hasattr(self, "evt_actuatorBumpTestStatus"):
+            await self.evt_actuatorBumpTestStatus.set_write(
+                actuator=actuator, status=status.value
+            )
+
+    async def _do_killActuatorBumpTest(self, data: salobj.BaseMsgType) -> None:
+        """Kill the running actuator bump test in the closed-loop control.
+
+        Parameters
+        ----------
+        data : `object`
+            Data of the SAL message.
+        """
+
```

Comment:
I think you need to add an `self.assert_enabled()` here.

also, consider making it an immediate return (no `IN_PREOGRESS`) if the task is already cancelled;

```
if self._task_bump_test.done():
    return
```

Reply:
Done in the commit: https://github.com/lsst-ts/ts_m2/commit/db3ca385e39a6b5322b55a523e32a1b4d19a539f. Thanks!